rust/parser/test/mod.rs

```diff
@@ -427,7 +416,144 @@ $x isa movie,
     has tmdb-vote-count <= 400;"#;
 
     let parsed = parse_query(query).unwrap().into_match();
-    let expected = typeql_match!(var_concept("x").isa("movie").has(("tmdb-vote-count", lte(400))));
+    let expected = typeql_match!(cvar("x").isa("movie").has(("tmdb-vote-count", lte(400))));
+
+    assert_valid_eq_repr!(expected, parsed, query);
+}
+
+#[test]
+fn test_attribute_query_by_value_variable() {
+    let query = r#"match
+?x = 5;
+$a == ?x isa age;"#;
+
+    let parsed = parse_query(query).unwrap().into_match();
+    let expected = typeql_match!(vvar("x").assign(Expression::Constant(5.into())), cvar("a").eq(vvar("x")).isa("age"),);
```

Comment:
```suggestion
    let expected = typeql_match!(vvar("x").assign(5), cvar("a").eq(vvar("x")).isa("age"));
```

Reply:
We can create a function `constant` like as `cvar` and `vvar`.